File: Python/SelectionSort.py
#SelectionSort
def SelectionSort(l):
    for a in range(len(l)):
        #Take current id
        minId=a
        
        #Go throuw array to find min id
        for i in range(a,len(l)-1):
            if l[minId]>l[i+1]:
                minId=i+1
        
        #Swop midId and current Element
        l[a],l[minId]=l[minId],l[a]
    return l
ourList=[2,4,1,3,7,4,5]
print(SelectionSort(ourList))

File: Js/SelectionSort.js
const SelectionSort = (l) => {
  let sortedL = [...l];
  for (let i = 0; i < l.length - 1; i++) {
    let minIndex = i;
    for (let j = i + 1; j < l.length; j++) {
      if (sortedL[minIndex] > sortedL[j]) {
        minIndex = j;
      }
    }
    [sortedL[minIndex], sortedL[i]] = [sortedL[i], sortedL[minIndex]];
  }
  return sortedL;
};

const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(SelectionSort(l));
